File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/FieryRegeneration.lua
function GetName()
  return "Fiery Regeneration"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TurnTriggered }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint ~= TurnTriggered then
    return
  end

  char:Heal(char:GetMaxHP() * 0.2 * (GetStats(char):GetTotalStat(FIR) / 100))
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/FieryRegeneration.cs
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Afflictions
{
    public class FieryRegeneration : Affliction
    {
        public override string Name { get { return "Fiery Regeneration"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TurnTriggered, 
                };
            }
        }

        protected override void OnTurnTriggered(ICharacter character)
        {
            character.Heal(character.Stats.GetTotalStat(BaseStat.HP) * .2 * (character.Stats.GetTotalStat(EleMastery.FIR) / 100));
        }
    }
}